src/main/java/com/google/step/youtube/Channel.java

```diff
@@ -0,0 +1,60 @@
+package com.google.step.youtube;
+
+import java.util.Date;
+
+/**
+ * Object for a YouTube channel and information on its videos. 
+ */
+public class Channel {
```

Comment:
Read up and use @AutoValue library for any Value Types like this one so equals(), hashCode() work correctly from the start and you don't need to define/maintain them.

(There's also other features to generate Builders, so def read up on all that it can do, which is a lot.)

https://github.com/google/auto/blob/master/value/userguide/index.md

You might have to do some tweaking to your maven configs to get the generated code into your build.  Hopefully there's docs on that, otherwise I would ask the Intern chat since I'm sure others have tried to get AutoValue working.

Reply:
Changed to autovalue.